mac/scripts/build.sh

```diff
@@ -77,6 +86,67 @@ echo "Architecture: ARM64 only"
 # Clean build directory only if it doesn't exist
 mkdir -p "$BUILD_DIR"
 
+# Filtering function for reduced context output
+filter_build_output() {
+    # Filter out verbose lines while preserving errors, warnings, and important status
+    awk '
```

Comment:
I do worry that this is expensive for 10.000 * 2 invocations, it's a new process that runs each time.